gui/wxpython/vdigit/toolbars.py

```diff
@@ -858,6 +866,62 @@ def OnSelectMap(self, event):
 
         event.Skip()
 
+    def _topoZeroLengthCheck(self, map_name):
+        """Topology check for lines or boundaries of zero length
+
+        :param str map_name: full vector map name
```

Comment:
The code in this function should probably be a separate module. It should not be part of GUI because it is data processing. In general, we should be moving any processing and analytical code from GUI. The idea is to keep the GUI as small as possible while making all functionality available in GUI also available for scripting ideally in command line or at least in Python. This also allows for easier testing of the functionality both in terms of writing tests and manual testing in case of debugging an issue.